### PR TITLE
Replace Constants.installationId with a generated device id

### PR DESCRIPTION
**Pull request by hamudi-ahmed**
## Goal

Replaces the usage of `Constants.installationId` (which has been removed in SDK 51) with our own generated device id.

## Design

The device id is now generated using `@bugsnag/cuid` and is stored/retrieved synchronously on start using `expo-secure-store`.

This means there is now an additional peer dependency on `expo-secure-store` that must be installed alongside `@bugsnag/expo`

## Changeset

- added `@bugsnag/cuid` dependency and `expo-secure-store` peer dependency to `plugin-expo-device`
- replaced `Constants.installationId` with our own generated device id
- updated `bugsnag-expo-cli` to automatically install the `expo-secure-store` peer dependency
- added `UPGRADING.md`

## Testing

- added new unit tests
- reinstated skipped device.id tests and added new test to assert that the device id is persisted across app starts

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
 # Changelog
 
+## [Unreleased]
+
+## Breaking Changes
+
+This release contains an additional peer dependency on `expo-secure-store`. Run the bugsnag-expo-cli to add this dependency to your project:
```
Comment (hamudi-ahmed) [inline CHANGELOG.md]: I've followed the approach we used here for this: https://github.com/bugsnag/bugsnag-expo/blob/v50/next/CHANGELOG.md#v4400-2022-04-19

but is it better off in an `UPGRADING.md`?

Comment (gingerbenw) [reply CHANGELOG.md]: I like the sound of an upgrade guide and keeping code snippets out of the changelog 👍🏻

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,12 +2,17 @@
...
+  - [Reporting issues](#reporting-issues)
+    - [Fixing issues](#fixing-issues)
+    - [Adding features](#adding-features)
+  - [System requirements](#system-requirements)
+  - [Testing](#testing)
+  - [CI](#ci)
+  - [Keeping dependencies in sync](#keeping-dependencies-in-sync)
+  - [Updating the CLI to install a compatible notifier version](#updating-the-cli-to-install-a-compatible-notifier-version)
+  - [Releases](#releases)
+    - [Prereleases](#prereleases)
```
Comment (gingerbenw) [inline CONTRIBUTING.md:15]: 👌🏻

**Review by gingerbenw — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
 # Changelog
 
+## [Unreleased]
+
+## Breaking Changes
```
Comment (tomlongridge) [inline CHANGELOG.md:5]:
```suggestion

This release adds support for Expo 51.

## Breaking Changes
```

File: UPGRADING.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Upgrading
+
+## v50.x to v51.x
+
+This release contains an additional peer dependency on `expo-secure-store`. Run the bugsnag-expo-cli to add this dependency to your project:
```
Comment (tomlongridge) [inline UPGRADING.md:5]:
```suggestion

This release contains an additional peer dependency on `expo-secure-store` for persistence of a device ID between app launches (previously provided by the deprecated `Constants.installationId` field) for User Stability calculations.

Run the bugsnag-expo-cli to add this dependency to your project:
```

File: packages/plugin-expo-device/device.js [hunk 5]
```diff
@@ -2,6 +2,10 @@ const Device = require('expo-device')
 const Constants = require('expo-constants').default
 const { Dimensions, Platform } = require('react-native')
 const rnVersion = require('react-native/package.json').version
+const cuid = require('@bugsnag/cuid')
+const SecureStore = require('expo-secure-store')
+
+const DEVICE_ID_KEY = 'bugsnag-anonymous-id'
```
Comment (tomlongridge) [inline packages/plugin-expo-device/device.js:8]: Is there a reason not to call this `bugsnag-device-id`? I'm not sure if the name is at all visible externally though...

**Pull request merged**